### PR TITLE
chore: add Makefile and `codespell`

### PR DESCRIPTION
**Pull request by zerosnacks**
## Motivation

Adds a Makefile to help run the various command locally more easily

`$ make` yields:

```
Usage:
  make <target>

Help
  help             Display this help.

Build
  build            Build the project.

Other
  clean            Clean the project.
  fmt              Run all formatters.
  lint             Run all linters.
  test             Run all tests.
  pr               Run all tests and linters in preparation for a PR.
```

Includes common tasks focused on commands you would run for a PR

Adds `codespell` to Makefile and CI (also used by Reth) to enforce spell checking

## Solution

No breaking changes, fixes spelling issues raised by codespell

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
typo farmers in shambles

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Heavily inspired by:
+# - Lighthouse: https://github.com/sigp/lighthouse/blob/693886b94176faa4cb450f024696cb69cda2fe58/Makefile
+# - Reth: https://github.com/paradigmxyz/reth/blob/1f642353ca083b374851ab355b5d80207b36445c/Makefile
+.DEFAULT_GOAL := help
+
+# Cargo profile for builds.
+PROFILE ?= release
```
Comment (DaniPopes) [inline Makefile:7]: This should not default to release for 99% of the use cases of this file

File: crates/evm/fuzz/src/strategies/param.rs [hunk 2]
```diff
@@ -218,12 +218,12 @@ mod tests {
         let func = get_func(f).unwrap();
         let db = CacheDB::new(EmptyDB::default());
         let state = EvmFuzzState::new(&db, FuzzDictionaryConfig::default());
-        let strat = proptest::prop_oneof![
+        let strategy = proptest::prop_oneof![
```
Comment (DaniPopes) [inline crates/evm/fuzz/src/strategies/param.rs:221]: strat should be ignored

File: .codespellrc [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[codespell]
+skip = .git,target,testdata,Cargo.toml,Cargo.lock
+ignore-words-list = crate,ser,ratatui,Caf,froms
```
Comment (DaniPopes) [inline .codespellrc:3]: pls add new line :)

**Review by ArkFunds — APPROVED** [review]

**Review by ArkFunds — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (zerosnacks) [reply Makefile:7]: Should we change it to `dev`?

https://doc.rust-lang.org/cargo/reference/profiles.html#dev

Users can indeed override this with `PROFILE=release make build` if they need a release build

**Review by DaniPopes — APPROVED** [review]

**Pull request merged**